formats/polus-czi-extract-plugin/src/main.py

```diff
@@ -1,18 +1,189 @@
+def extract_fovs(file_path: Path,
+                 out_path: Path):
+    """ Extract individual FOVs from a czi file
+
+    When CZI files are loaded by BioFormats, it will generally try to mosaic
+    images together by stage position if the image was captured with the
+    intention of mosaicing images together. At the time this function was
+    written, there was no clear way of extracting individual FOVs so this
+    algorithm was created.
+    
+    Every field of view in each z-slice, channel, and timepoint contained in a
+    CZI file is saved as an individual image.
+
+    Args:
+        file_path (Path): Path to CZI file
+        out_path (Path): Path to output directory
+    """
+    
+    with ProcessManager.process(file_path.name):
+        
+        logger.info('Starting extraction from ' + str(file_path) + '...')
+
+        if Path(file_path).suffix != '.czi':
+            TypeError("Path must be to a czi file.")
+            
+        base_name = Path(file_path.name).stem
+        
+        # Load files without mosaicing
+        czi = czifile.CziFile(file_path,detectmosaic=False)
+        subblocks = [s for s in czi.filtered_subblock_directory if s.mosaic_index is not None]
+        
+        ind = {'X': [],
+               'Y': [],
+               'Z': [],
+               'C': [],
+               'T': [],
+               'Row': [],
+               'Col': []}
+        
+        # Get the indices of each FOV
+        for s in subblocks:
+            scene = [dim.start for dim in s.dimension_entries if dim.dimension=='S']
+            if scene is not None and scene[0] != 0:
+                continue
+            
+            for dim in s.dimension_entries:
+                if dim.dimension=='X':
+                    ind['X'].append(dim.start)
+                elif dim.dimension=='Y':
+                    ind['Y'].append(dim.start)
+                elif dim.dimension=='Z':
+                    ind['Z'].append(dim.start)
+                elif dim.dimension=='C':
+                    ind['C'].append(dim.start)
+                elif dim.dimension=='T':
+                    ind['T'].append(dim.start)
+                    
+        row_conv = {y:row for (y,row) in zip(np.unique(np.sort(ind['Y'])),range(0,len(np.unique(ind['Y']))))}
+        col_conv = {x:col for (x,col) in zip(np.unique(np.sort(ind['X'])),range(0,len(np.unique(ind['X']))))}
+        
+        ind['Row'] = [row_conv[y] for y in ind['Y']]
+        ind['Col'] = [col_conv[x] for x in ind['X']]
+        
+        with BioReader(file_path) as br:
+            
+            metadata = br.metadata
+            chan_names = br.cnames
+        
+        for s,i in zip(subblocks,range(0,len(subblocks))):
+            
+            Z = None if len(ind['Z'])==0 else ind['Z'][i]
+            C = None if len(ind['C'])==0 else ind['C'][i]
+            T = None if len(ind['T'])==0 else ind['T'][i]
+        
+            out_file_path = out_path.joinpath(_get_image_name(base_name,
```

Comment:
Your question is definitely clear, but this code is performing as expected.

What this code does is directly scrape metadata from the native czi format. In this metadata, individual fields of view have positional information in XYZCT format, although it's different than what we have in bfio. Line 145 is looking for the presence of Z in the metadata of the CZI file, and setting it to `None` if it's not present. For the purposes of file names, if `Z is None`, that means there are not different z-slices in the input file so no modification to the file name is required. If there is a Z value in the metadata, then we add an `_Z{}` to the file name to indicate the z-position associated with the output file.

The `_get_image_name` should always have the Z-size set to 1 since all images are exported as 2d images. This is done to keep the same output format as the tiled tiff converter plugin, and was something we decided with NIST should be the standard import and conversion format. So, all image have to have a Z-size of one.